### ISSUE TITLE
Python 3.4: `pip install git+git://github.com/hyperopt/hyperopt.git` fails

### ISSUE THREAD

**Issue by albertz** [hyperopt/hyperopt#231]
```
...
    copying launcher.c -> build/src
    Skipping implicit fixer: buffer
    Skipping implicit fixer: idioms
    Skipping implicit fixer: set_literal
    Skipping implicit fixer: ws_comma
    Traceback (most recent call last):
      File "/tmp/tmpo9qh0qhq/distribute-0.6.38/build/src/pkg_resources.py", line 2465, in _dep_map
        return self.__dep_map
      File "/tmp/tmpo9qh0qhq/distribute-0.6.38/build/src/pkg_resources.py", line 2290, in __getattr__
        raise AttributeError(attr)
    AttributeError: _DistInfoDistribution__dep_map
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "setup.py", line 248, in <module>
        scripts = scripts,
      File "/u/beck/programs/python/3.4.3-static/lib/python3.4/distutils/core.py", line 108, in setup
        _setup_distribution = dist = klass(attrs)
      File "/tmp/tmpo9qh0qhq/distribute-0.6.38/build/src/setuptools/dist.py", line 225, in __init__
        _Distribution.__init__(self,attrs)
      File "/u/beck/programs/python/3.4.3-static/lib/python3.4/distutils/dist.py", line 280, in __init__
        self.finalize_options()
      File "/tmp/tmpo9qh0qhq/distribute-0.6.38/build/src/setuptools/dist.py", line 257, in finalize_options
        ep.require(installer=self.fetch_build_egg)
      File "/tmp/tmpo9qh0qhq/distribute-0.6.38/build/src/pkg_resources.py", line 2028, in require
        working_set.resolve(self.dist.requires(self.extras),env,installer)))
      File "/tmp/tmpo9qh0qhq/distribute-0.6.38/build/src/pkg_resources.py", line 2238, in requires
        dm = self._dep_map
      File "/tmp/tmpo9qh0qhq/distribute-0.6.38/build/src/pkg_resources.py", line 2467, in _dep_map
        self.__dep_map = self._compute_dependencies()
      File "/tmp/tmpo9qh0qhq/distribute-0.6.38/build/src/pkg_resources.py", line 2500, in _compute_dependencies
        common = frozenset(reqs_for_extra(None))
      File "/tmp/tmpo9qh0qhq/distribute-0.6.38/build/src/pkg_resources.py", line 2497, in reqs_for_extra
        if req.marker_fn(override={'extra':extra}):
      File "/tmp/tmpo9qh0qhq/distribute-0.6.38/build/src/_markerlib/markers.py", line 109, in marker_fn
        return eval(compiled_marker, environment)
      File "<environment marker>", line 1, in <module>
    NameError: name 'sys_platform' is not defined
    /tmp/pip-2t9mt000-build/distribute-0.6.38-py3.4.egg
    Traceback (most recent call last):
      File "/tmp/pip-2t9mt000-build/distribute_setup.py", line 150, in use_setuptools
        raise ImportError
    ImportError
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 20, in <module>
      File "/tmp/pip-2t9mt000-build/setup.py", line 38, in <module>
        distribute_setup.use_setuptools()
      File "/tmp/pip-2t9mt000-build/distribute_setup.py", line 152, in use_setuptools
        return _do_download(version, download_base, to_dir, download_delay)
      File "/tmp/pip-2t9mt000-build/distribute_setup.py", line 132, in _do_download
        _build_egg(egg, tarball, to_dir)
      File "/tmp/pip-2t9mt000-build/distribute_setup.py", line 123, in _build_egg
        raise IOError('Could not build the egg.')
    OSError: Could not build the egg.
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-2t9mt000-build
```

**Comment by albertz** [comment]
My distribute, setuptools, etc packages are all up-to-date.

**Comment by temporaer** [comment]
I get the same error on a python2.7 installation (w/o the fix) on a Ubuntu vivid system. Could it be due to the fact that distribute has been merged back to setuptools and is now deprecated?

**Comment by renewang** [comment]
I have the same problem. But I installed from git clone repository. And when executing 
`python setup.py develop --user `
with python3.4 and I got several errors in some source codes as following:
**Can't parse build/lib/hyperopt/base.py: ParseError: bad input: type=22, value='=', context=('', (312, 64))**
I found the setup script will download distribute and installed. 
So I removed distribute_setup.py from repository and install again. 
setup.py script will complain distribute_setup.py not found and use system setuptools
At this time, installation is successful and no parse errors occur. However, I now in another new problem stated in #234 (AttributeError: 'function' object has no attribute 'fmin') when try to use hyperopt.fmin.fmin function.
